lib/Twig/NodeVisitor/Optimizer.php

```diff
@@ -45,6 +46,7 @@ public function __construct($optimizers = -1)
         }
 
         $this->optimizers = $optimizers;
+        $this->isPHP540RCorHigher = version_compare(phpversion(), '5.4.0RC1', '>=');
```

Comment:
I suggest bumping the requirement to 5.2.7 instead, to be able to use the constant all the time (which will also avoid using a property in the class)